src/docker.rs

```diff
@@ -206,9 +192,47 @@ pub fn run(
         }
     }
 
+    // We need to specify the user for Docker, but not for Podman.
+    let uid_gid = if let Ok(ce) = get_container_engine() {
+        if ce.ends_with(DOCKER) {
+            Some((
+                env::var("CROSS_CONTAINER_UID").unwrap_or(id::user().to_string()),
+                env::var("CROSS_CONTAINER_GID").unwrap_or(id::group().to_string()),
+            ))
+        } else {
+            None
+        }
+    } else {
+        None
+    };
```

Comment:
Please move the code back to the original place as it is a non-required code change.